### ISSUE TITLE
Errors not reported on properties inside function calls

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#25695]
**TypeScript Version:**  3.0.0-dev.201xxxxx

**Search Terms:** 

The changes from https://github.com/Microsoft/TypeScript/pull/25140, did not seem to work on function calls.

**Code**

```ts
type Options = { x: number, y: string };
declare function foo(options: Options): void;

foo({ x: 1, y: 10 });
//  ^^^^^^^^^^^^^^^
// Argument of type '{ x: number; y: number; }' is not assignable to parameter of type 'Options'.
//   Types of property 'y' are incompatible.
//    Type 'number' is not assignable to type 'string'.


var m: Options;
m = { x: 1, y: 2 };
//          ^
//  Type 'number' is not assignable to type 'string'.
```

**Expected behavior:**

**Actual behavior:**

**Playground Link:**  

**Related Issues:**

**Comment by CyrusNajmabadi** [comment]
hey Mohamed/Dan/Ryan :)   This is definitely affecting us over here.  Many of our APIs are intended to be used as factory constructors, where the construction values are passed in through large complex objects.  

We'd definitely like get the new minimal-error experience as it's really much better for our domain.  Just adding a +1 to having this improved in an upcoming release.  Thanks!

**Comment by RyanCavanaugh** [comment]
👋 !
